packages/components/src/components/SquiggleEditorImportedBindings.tsx

```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import { SquiggleEditor } from "./SquiggleEditor";
+import type { SquiggleEditorProps } from "./SquiggleEditor";
+import { runPartial, defaultBindings } from "@quri/squiggle-lang";
+import type { result, errorValue, bindings } from "@quri/squiggle-lang";
+
+function resultDefault(x: result<bindings, errorValue>): bindings {
+  switch (x.tag) {
+    case "Ok":
+      return x.value;
+    case "Error":
+      return defaultBindings;
+  }
+}
+
+function replaceBindings(
+  props: SquiggleEditorProps,
+  newBindings: bindings
+): SquiggleEditorProps {
+  return { ...props, bindings: newBindings };
+}
+
+export type SquiggleEditorImportedBindingsProps = SquiggleEditorProps & {
+  bindingsImportFile: string;
+};
+
+export const SquiggleEditorImportedBindings: React.FC<
+  SquiggleEditorImportedBindingsProps
+> = (props) => {
+  const [bindingsResult, setBindingsResult] = React.useState({
+    tag: "Ok",
+    value: defaultBindings,
+  } as result<bindings, errorValue>);
+  React.useEffect(() => {
+    async function retrieveBindings(fileName: string) {
+      let contents = await fetch(fileName).then((response) => {
+        return response.text();
+      });
+      setBindingsResult(runPartial(contents));
+    }
+    retrieveBindings(props.bindingsImportFile);
+  }, []);
+  const deliveredBindings = resultDefault(bindingsResult);
+  const newProps = replaceBindings(props, deliveredBindings);
```

Comment:
Minor, but: separate function here seems excessive, I'd just inline `{...props, bindings: bindingsResult }`.

Reply:
Also: can you clean out the `bindingsImportFile` property out of `props` you pass to `SquiggleEditor`?

E.g., with `const { bindingsImportFile, ...editorProps } = props;` at the start of the component. It doesn't cause any issues, but still, would be clearer (e.g., I was looking for `SquiggleEditor` in react dev tools and extra prop made it harder to identify the correct component.

<img width="782" alt="Screenshot 2022-07-28 at 17 19 20" src="https://user-images.githubusercontent.com/89368/181515051-5559ca41-d279-440f-900f-bcb2b34ec378.png">

Reply:
This is causing a major bork :/ 

![2022-07-28_10-15](https://user-images.githubusercontent.com/39039420/181531849-e6764e7a-b87b-4fad-a25f-0fb37195f2b5.png)

Reply:
You need double braces:
```jsx
<SquiggleEditor {...{...editorProps, bindings: deliveredBindings}} />
```

(Outer `{...` is JSX syntax, not ES6 syntax)